webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -498,6 +488,24 @@ def _define_style_ensemble(index: int, x_axis: pd.DataFrame) -> Dict:
     return style
 
 
+def _update_corr_index_dict(
+    response_x_axis: Union[pd.Index, list], observation_x_axis: pd.Index
```

Comment:
We say that we expect `response_x_axis` to be either a `pd.Index` or `list`, yet later we check if parameter is one of the two. What would happen if user/developer calles function using neither options? `updated_index` is not even assigned in that case. This is me, but I tend to use `if` conditions for expects that are not part of the core functionality of the function. This will reduce (nested) `if` conditions, reduce the complexity, reduce chances of unassignment and as a side effect make the code more readable.
example:
```python
def _update_corr_index_dict(
    response_x_axis: Union[pd.Index, list], observation_x_axis: pd.Index
) -> int:
    if type(response_x_axis) not in (pd.Index, list):
        raise ValueError
    return {pd.Index: responds_x_axis.get_loc, list: response_x_axis.index}.get(response_x_axis)(_get_first_observation_x(observation_x_axis))
#--------readability----------
x_axis_default_observation = _get_first_observation_x(observation_x_axis)
func_selector = {pd.Index: responds_x_axis.get_loc, list: response_x_axis.index}
return func_selector.get(response_x_axis)(x_axis_default_observation)
```

Reply:
Cool suggestion @sregales-TNO but I am not sure I agree with everything suggested. 

I agree that adding a check on the expected types for the response_x_axis is a good idea with raising an error or implementing specific handling when we get unexpected types for `response_x_axis`. 

But I don't encourage using a map indexed on data types, looks a bit too funcky for my taist. 

also the suggested solution does not seem to work as it is, it requires replacing `response_x_axis` with `type(response_x_axis)`:

``` python
x_axis_default_observation = _get_first_observation_x(observation_x_axis) func_selector = {pd.Index: responds_x_axis.get_loc, list: response_x_axis.index} return func_selector.get(type(response_x_axis))(x_axis_default_observation)
```
and in my opinion, this is no more readable than just than what it was there before :shrug:

Reply:
thanks for the correction. yeah, I guess it's a matter of preference, I normally swap to mappings when it's multiple consecutive conditional statements on the same element/object with slightly different behavior. I guess for two conditionals it seems overkill. A better use would be 3+ consecutive ones